File: frontend/src/app/core/interfaces/find-number-game.dto.ts
import { GAME_DIFFICULTY_LEVEL, GAME_STATUSES } from '../enums';

export interface FindNumberGameDTO {
  id: string;
  player: string;
  totalNumbersToFind: number;
  difficultyLevel: GAME_DIFFICULTY_LEVEL;
  startTime: Date;
  endTime: Date;
  completionTime: number;
  gameStatus: GAME_STATUSES;
  lastSelectedNumber: number;
  elapsedTime: number;
  bonusTime: number;
  updateAt: Date;
}

export interface FindNumberGameBestRecordDTO {
  gameId: string;
  playerFullName: string;
  playerDisplayname: string;
  playerUsername: string;
  archivedTime: string;
  completionTime: number;
}

export interface FindNumberGameHistoryItem {
  startTime: string;
  endTime: string;
  completionTime: number;
  bonusTime: number;
}

export interface FindNumberGameHistory {
  history: FindNumberGameHistoryItem[];
  bestRecord: FindNumberGameBestRecordDTO;
  unfinishedGames: FindNumberGameDTO[];
  totalGamesPlayed: number;
}

export interface FindNumberGameHistory {
  history: FindNumberGameHistoryItem[];
  bestRecord: FindNumberGameBestRecordDTO;
}

File: backend/src/main/java/com/nguyen_trong_nhat/you_and_i/features/games/find_number_game/dto/FindNumberGameDTO.java
package com.nguyen_trong_nhat.you_and_i.features.games.find_number_game.dto;

import com.nguyen_trong_nhat.you_and_i.features.user.entity.MyUserDetail;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;

import java.time.Instant;
import java.util.UUID;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class FindNumberGameDTO {
    private UUID id;

    private String player;

    private Integer totalNumbersToFind;

    private DifficultyLevel difficultyLevel;

    private Instant startTime;

    private Instant endTime;

    private String completionTime;

    private GameStatus gameStatus;

    private Integer lastSelectedNumber;

    private String  elapsedTime;

    private Integer bonusTime;

    private Instant updateAt;
}
